patches/net/minecraft/server/level/ServerPlayer.java.patch

```diff
@@ -146,16 +146,31 @@
              }
  
              this.nextContainerCounter();
-@@ -1222,10 +_,16 @@
+@@ -1222,10 +_,31 @@
  
                  return OptionalInt.empty();
              } else {
-+                if (extraDataWriter == null) {
++                var extraData = net.neoforged.neoforge.common.util.FriendlyByteBufUtil.writeCustomData(
++                        buffer -> {
++                            p_9033_.writeClientSideData(abstractcontainermenu, buffer);
++                            if (extraDataWriter != null) {
++                                extraDataWriter.accept(buffer);
++                            }
++                        },
++                        registryAccess()
++                );
++                if (extraData.length == 0) {
```

Comment:
Very particular nitpick: I would invert this in order to keep the additions in a single block